src/VirtoCommerce.DescriptionExportImportModule.Data/Services/CsvPagedEditorialReviewDataImporter.cs

```diff
@@ -0,0 +1,168 @@
+
+                    var productsForReviewAdding = productSearchResult.Results;
+
+                    AddReviewsToProducts(productsForReviewAdding, importReviewsForAdding);
+
+                    allProductsForSaving.AddRange(productsForReviewAdding);
+                }
+
+                var importReviewsForUpdating = importReviewRecords.Where(x => !string.IsNullOrEmpty(x.Record.Id))
+                    .Select(x => x.Record).ToArray();
+
+                if (importReviewsForUpdating.Length > 0)
+                {
+                    var productIds = existedReviews.Select(x => x.ItemId).ToArray();
+
+                    var productSearchResult = await _productSearchService.SearchProductsAsync(
+                        new ProductSearchCriteria() { ObjectIds = productIds, ResponseGroup = ItemResponseGroup.ItemEditorialReviews.ToString() });
+
+                    var productsForReviewUpdating = productSearchResult.Results;
+
+                    PatchProductsReviews(productsForReviewUpdating, importReviewsForUpdating);
+
+                    allProductsForSaving.AddRange(productsForReviewUpdating);
+                }
+
+                await _itemService.SaveChangesAsync(allProductsForSaving.ToArray());
+
+
+                importProgress.CreatedCount += importReviewsForAdding.Length;
+                importProgress.UpdatedCount += importReviewsForUpdating.Length;
+            }
+            catch (Exception e)
+            {
+                HandleError(progressCallback, importProgress, e.Message);
+            }
+            finally
+            {
+                importProgress.ProcessedCount = Math.Min(dataSource.CurrentPageNumber * dataSource.PageSize,
+                    importProgress.TotalCount);
+                importProgress.ErrorCount = importProgress.ProcessedCount - importProgress.CreatedCount -
+                                            importProgress.UpdatedCount;
+            }
+        }
+
+
+        private static void SetIdAndSkuToNullByExistence(ImportRecord<CsvEditorialReview>[] importReviewRecords, EditorialReview[] existedReviews)
+        {
+            foreach (var importRecord in importReviewRecords)
+            {
+                if (existedReviews.Any(x => x.Id == importRecord.Record.Id))
+                {
+                    importRecord.Record.ProductSku = null;
+                }
+                else
+                {
+                    importRecord.Record.Id = null;
+                }
+            }
+        }
+
+        private static void AddReviewsToProducts(IList<CatalogProduct> productsForReviewAdding, CsvEditorialReview[] importReviewsForAdding)
+        {
+            foreach (var product in productsForReviewAdding)
+            {
+                var productImportReviews =
+                    importReviewsForAdding.Where(x => x.ProductSku.EqualsInvariant(product.Code)).ToArray();
```

Comment:
Maybe put this LINQ into the foreach statement

Reply:
How?

Reply:
These are different collections.

Reply:
```suggestion
                    
```